src/renderer/helpers/addressHelper.ts

```diff
@@ -0,0 +1,20 @@
+import { BNBChain, CosmosChain, THORChain } from '@xchainjs/xchain-util'
+
+export const truncateAddress = (addr: string, chain: string, network: string): string => {
+  const first = addr.substr(0, getAddressPrefixLength(chain, network) + 3)
+  const last = addr.substr(addr.length - 3, 3)
+  return `${first}...${last}`
+}
+
+export const getAddressPrefixLength = (chain: string, network: string): number => {
+  switch (chain) {
+    case BNBChain:
+      return network === 'testnet' ? 4 : 3
+    case THORChain:
+      return network === 'testnet' ? 5 : 4
```

Comment:
Unfortunately `xchain-thorchain` [does not export `getPrefix` - it still private](https://github.com/xchainjs/xchainjs-lib/blob/master/packages/xchain-thorchain/src/client.ts#L232-L234). Let's change it (see https://github.com/xchainjs/xchainjs-lib/issues/183)